catalogd/internal/serverutil/serverutil.go

```diff
@@ -42,17 +45,22 @@ func AddCatalogServerToManager(mgr ctrl.Manager, cfg CatalogServerConfig, tlsFil
 	}
 
 	shutdownTimeout := 30 * time.Second
+	handler := cfg.LocalStorage.StorageServerHandler()
+	handler = gzhttp.GzipHandler(handler)
+	handler = catalogdmetrics.AddMetricsToHandler(handler)
 
 	l := mgr.GetLogger().WithName("catalogd-http-server")
-	handler := catalogdmetrics.AddMetricsToHandler(cfg.LocalStorage.StorageServerHandler())
 	handler = logrLoggingHandler(l, handler)
 
 	catalogServer := manager.Server{
 		Name:                "catalogs",
 		OnlyServeWhenLeader: true,
 		Server: &http.Server{
-			Addr:        cfg.CatalogAddr,
-			Handler:     handler,
+			Addr:    cfg.CatalogAddr,
+			Handler: handler,
+			BaseContext: func(_ net.Listener) context.Context {
+				return log.IntoContext(context.Background(), mgr.GetLogger().WithName("http.catalogs"))
+			},
```

Comment:
I _think_ we can either drop this whole `BaseContext` or alternately, simply `return l`

Reply:
Either way, I don't think the code uses the logger from the context. This was a holdover from the way that the original logging middleware worked.

Reply:
Dropped it.